---
title: "User Manual - EnrichTool"
author: "Martin TÃ¶rn"
date: "2025-12-12"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Enrichment analysis using EnrichTool 

In manual will walk you though the steps required to perform enrichment analysis on a case study dataset.

This package use functions from R-Packages which needs to be installed:

* edgeR
* clusterProfiler
* enrichPlot
* org.Hs.eg.db
* openxlsx

The manual is divded into parts where each function is explained:

* Importing the data
* Identification of differentially expressed genes
* Exporting results
* Over representation analysis (ORA)
* Visualization of ORA results

### Importing the data

The RNA-seq count data can be imported with the function *import_RNA()* which needs the path to your count data and has a default filter of rows by mean log2 cpm > 1. This can be changed by setting the parameter *cutoff = int*.

```{r import_RNA() code}
library(EnrichTool)
#Import count data and filter on log 2 cpm
count_table <- import_RNA("Examples/E-MTAB-2523.counts.txt")
```

### Identification of differentially expressed genes

To identify differentially expressed gene, the function *degAnalysis()* can be used. When using this function you need to setup desired design and contrasts for your study. The example below is a partially paired case study which means that we create one design matrix containing the information about sample condition and also which id each sample corrosponds to. The function also needs a contrast matrix which should reflect the desired groups to be compared. These arguments can be passed to the function along with the filtered count data generated from *import_RNA()*.

```{r degAnalysis code}
#Extract the sample information from sdrf file
sdrf <- data.table::fread("Examples/E-MTAB-2523_sample table.txt")

#Set the group variable and make "healthy" baseline
Patient <- factor(sdrf$individual)
Sample <- sdrf$sample
Disease <- factor(sdrf$disease, levels = c("normal", "carcinoma"))

SampleTable <- data.frame(Sample, Disease, Patient)
SampleTable <- data.frame(SampleTable, row.names = 1)
print(head(SampleTable))

#Define design matrix
design <- model.matrix(~ 0 + Disease + Patient, data = SampleTable)
colnames(design)[1:2] <- c("normal", "carcinoma")
#define contrast matrix
contrastmat <- limma::makeContrasts(carsinomaVsnormal = carcinoma - normal,
                             levels = design)
print(head(contrastmat))

#Run DEG analysis on the filtered count data and provide the group variable
deg_res <- deg_analysis(count_table, design, contrastmat)
print(head(deg_res))
```

### Exporting results

Exporting the results from the DEG analysis is done with the function *exportToExcel()*. This function needs the result generated from previous function *degAnalysis()* and will then generate an excel file with DEGs.

The file name can be set using the parameter *file_name = "youfilename.xlsx"*, but it is not required.

```{r exportToExcel() code}
#export result
export_to_excel(deg_res)
```

### Over representation analysis (ORA)

Over representation analysis (ORA) is preformed using the function *ora_res* on the DEG results generated from previous function *degAnalysis()* and generates a list of results including enrichment analysis using GO and KEGG. 

The parameter *key_type* is by default set to ENSEMBL but can be altered to suit your need. In the example below the *key_type* is set to SYMBOL.

```{r ora_res() code, results='hide', message=FALSE, warning=FALSE}
#Run ORA (GO and KEGG) on the deg results
ora <- ora_res(deg_res, key_type = "SYMBOL")
```

### Visualization of ORA results

The results from ORA can be visualized using the function *ora_plot()* which can be used on the output from the previous function *ora_res()*. This will by default generate 2 dotplots and 2 cnetplots for GO and KEGG enrichment analysis.

If only one type of plot is desired the parameter *plots* can be set to either "dotplot" or "cnetplot".

```{r plot_ora() code, results='hide'}
#plot enrichment analysis
ora_plot(ora)
```

```{r, fig.width=5, fig.height=5, echo=FALSE}
knitr::include_graphics("cnet_go.png")

```

```{r, fig.width=6, fig.height=6, echo=FALSE}
knitr::include_graphics("dot_kegg.png")
```

